### PR TITLE
Move to LLVM 14

### PR DESCRIPTION
**Pull request by idavis**

### CODE REVIEW

**Review by swernli — COMMENTED** [review]
Might be good to update the devcontainer too, even though that needs other work/updates besides the LLVM version.

File: .github/workflows/trusted-cd-mac-release.yml [hunk 1]
```diff
@@ -17,14 +17,15 @@ jobs:
       - name: "Installing dependencies"
         run: |
           rm '/usr/local/bin/2to3'
-          brew install llvm@13
+          brew uninstall llvm@13
```
Comment (swernli) [inline .github/workflows/trusted-cd-mac-release.yml:20]: This one has an explicit uninstall of LLVM 13, but the ci-mac-on-main-pr.yml just installs 14. Is it necessary to have this uninstall here, and if so, why isn't it needed for the CI path?

**Review by swernli — APPROVED** [review]

File: qir/qat/Passes/StaticResourceComponent/AllocationAnalysisPass.cpp [hunk 2]
```diff
@@ -29,7 +29,7 @@ bool AllocationAnalysisPass::extractResourceId(llvm::Value* value, uint64_t& ret
             return false;
         }
 
-        llvm::Type* element_type = pointer_type->getElementType();
```
Comment (billti) [inline qir/qat/Passes/StaticResourceComponent/AllocationAnalysisPass.cpp:32]: Do we still need lines 26 to 30 above? There's no other use of `pointer_type` with this change. Or do you still need the null check on line 27 (i.e. you can't just check if element_type is null here)

**Review by billti — APPROVED** [review]

File: qir/qat/Passes/StaticResourceComponent/AllocationAnalysisPass.cpp (hunk 2 above)
Comment (idavis) [reply qir/qat/Passes/StaticResourceComponent/AllocationAnalysisPass.cpp:32]: @billti FYI, we eventually have to completely change this code for opaque pointers.

**Pull request merged**